eng/CodeAnalysis.ruleset

```diff
@@ -0,0 +1,385 @@
+    <Rule Id="SA1502" Action="Info" /> <!-- Element should not be on a single line --> 
+    <Rule Id="SA1503" Action="Info" /> <!-- Braces should not be omitted --> 
+    <Rule Id="SA1504" Action="Info" /> <!-- All accessors should be single-line or multi-line --> 
+    <Rule Id="SA1505" Action="Info" /> <!-- An opening brace should not be followed by a blank line --> 
+    <Rule Id="SA1506" Action="Info" /> <!-- Element documentation headers should not be followed by blank line --> 
+    <Rule Id="SA1507" Action="Info" /> <!-- Code should not contain multiple blank lines in a row --> 
+    <Rule Id="SA1508" Action="Info" /> <!-- A closing brace should not be preceded by a blank line --> 
+    <Rule Id="SA1509" Action="Info" /> <!-- Opening braces should not be preceded by blank line --> 
+    <Rule Id="SA1510" Action="Info" /> <!-- 'else' statement should not be preceded by a blank line --> 
+    <Rule Id="SA1512" Action="Info" /> <!-- Single-line comments should not be followed by blank line --> 
+    <Rule Id="SA1513" Action="Info" /> <!-- Closing brace should be followed by blank line --> 
+    <Rule Id="SA1514" Action="Info" /> <!-- Element documentation header should be preceded by blank line --> 
+    <Rule Id="SA1515" Action="Info" /> <!-- Single-line comment should be preceded by blank line --> 
+    <Rule Id="SA1516" Action="Info" /> <!-- Elements should be separated by blank line --> 
+    <Rule Id="SA1517" Action="None" />
+    <Rule Id="SA1519" Action="Info" /> <!-- Braces should not be omitted from multi-line child statement --> 
+    <Rule Id="SA1520" Action="Info" /> <!-- Use braces consistently --> 
+    <Rule Id="SA1600" Action="None" />
+    <Rule Id="SA1601" Action="Info" /> <!-- Partial elements should be documented --> 
+    <Rule Id="SA1602" Action="Info" /> <!-- Enumeration items should be documented --> 
+    <Rule Id="SA1604" Action="Info" /> <!-- Element documentation should have summary --> 
+    <Rule Id="SA1605" Action="Info" /> <!-- Partial element documentation should have summary --> 
+    <Rule Id="SA1606" Action="Info" /> <!-- Element documentation should have summary text --> 
+    <Rule Id="SA1608" Action="Info" /> <!-- Element documentation should not have default summary --> 
+    <Rule Id="SA1610" Action="Info" /> <!-- Property documentation should have value text --> 
+    <Rule Id="SA1611" Action="Info" /> <!-- The documentation for parameter 'message' is missing --> 
+    <Rule Id="SA1612" Action="Info" /> <!-- The parameter documentation is at incorrect position --> 
+    <Rule Id="SA1614" Action="Info" /> <!-- Element parameter documentation should have text --> 
+    <Rule Id="SA1615" Action="None" />
+    <Rule Id="SA1616" Action="Info" /> <!-- Element return value documentation should have text --> 
+    <Rule Id="SA1617" Action="None" />
+      <Rule Id="SA1618" Action="Info" /> <!-- The documentation for type parameter is missing --> 
```

Comment:
Nit: there are a few extra indents in this file.

Reply:
I've pushed through a fix to all the indent issues.